src/Kis/Kis.hs

```diff
@@ -31,11 +35,33 @@ data KisRequest a where
 
 deriving instance Show (KisRequest a)
 
+data RequestType = ReadRequest | WriteRequest
+
+requestType :: KisRequest a -> RequestType
+requestType (CreateBed _) = WriteRequest
+requestType (CreatePatient _) = WriteRequest
+requestType (PlacePatient _ _) = WriteRequest
+requestType (GetPatient _) = ReadRequest
+requestType GetPatients = ReadRequest
+
+toNotif :: KisRequest a -> Notification
+toNotif (CreateBed _) = Notification NewBed
+toNotif (CreatePatient _) = Notification NewPatient
+toNotif (PlacePatient _ _) = Notification PatientMoved
+toNotif _ = Notification NoNotif
+
 data Kis m =
     Kis
     { k_requestHandler :: forall a. KisRequest a -> m a
+      -- ^ Interface with the data
     , k_clock :: Clock
+      -- ^ Interface with clock functions
+    , k_getNotifications :: m [Entity Notification]
+    , k_waitForNewNotification :: m ()
+    , k_deleteNotification :: NotificationId -> m ()
+      -- ^ Interface to the notifications system
     }
```

Comment:
Dieser Typ wächst schonwieder und ich weiss nicht genau, ob das so gut ist. Vielleicht sollten die Notificationssachen zumindestens nochmal in einen eigenen Typen (GADT? ) unterteilt werden, was aber natürlich nichts daran ändert, dass man die Sachen alle irgendwie implementieren oder mocken muss, falls man das Kis aufbaut. `waitForNewNotification` unterscheided sich halt einbisschen von den anderen zwei, weil es ein wait of die MVar ist und keine Query auf die Datenbank.

Einen einfachen Weg das aufzuteilen sehe ich gerade auch nicht.